front/lib/resources/group_resource.ts

```diff
@@ -0,0 +1,134 @@
+import type { ModelId, Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import type { Attributes, ModelStatic, Transaction } from "sequelize";
+
+import type { Authenticator } from "@app/lib/auth";
+import type { Workspace } from "@app/lib/models/workspace";
+import { BaseResource } from "@app/lib/resources/base_resource";
+import { GroupModel } from "@app/lib/resources/storage/models/groups";
+import type { ReadonlyAttributesType } from "@app/lib/resources/storage/types";
+import { getResourceIdFromSId, makeSId } from "@app/lib/resources/string_ids";
+
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface, @typescript-eslint/no-unsafe-declaration-merging
+export interface GroupResource extends ReadonlyAttributesType<GroupModel> {}
+// eslint-disable-next-line @typescript-eslint/no-unsafe-declaration-merging
+export class GroupResource extends BaseResource<GroupModel> {
+  static model: ModelStatic<GroupModel> = GroupModel;
+
+  private workspace?: Workspace;
+
+  constructor(model: ModelStatic<GroupModel>, blob: Attributes<GroupModel>) {
+    super(GroupModel, blob);
+  }
+
+  static async makeNew(blob: Attributes<GroupModel>) {
+    const group = await GroupModel.create({
+      ...blob,
+    });
+
+    return new this(GroupModel, group.get());
+  }
+
+  get sId(): string {
+    return GroupResource.modelIdToSId({
+      id: this.id,
+      workspaceId: this.workspaceId,
+    });
+  }
+
+  static modelIdToSId({
+    id,
+    workspaceId,
+  }: {
+    id: ModelId;
+    workspaceId: ModelId;
+  }): string {
+    return makeSId("group", {
```

Comment:
I think flavien started a trend towards three letters prefixes (`fil` for files)

I'm neutral with small preference for full words. @flvndvd wdyt?

Reply:
should be `grp` as per the diff below in string_ids ?

Reply:
So as you seen the mapping group > grp is defined below on RESOURCES_PREFIX.
 I figured our "grp" was better than "gro" 😄

Reply:
Now I wonder why we have this mapping :) Looks good then :+1:

Reply:
Cause ideally we should do `GroupResource.className` 😛